src/front/glsl/parser.rs

```diff
@@ -503,22 +539,37 @@ pomelo! {
         (q,t)
     }
 
+    interpolation_qualifier ::= Interpolation((_, i)) {
+        i
+    }
+
     layout_qualifier ::= Layout LeftParen layout_qualifier_id_list(l) RightParen {
-        l
+        if let Some((_, loc)) = l.iter().find(|&q| q.0.as_str() == "location") {
+            Binding::Location(*loc)
+        } else if let Some((_, binding)) = l.iter().find(|&q| q.0.as_str() == "binding") {
```

Comment:
this is implicitly pattern matches by reference. The type on the right is `Option<&Something>`. So if you do this as `Some(&(_, binding))` you'd make it more explicit, clear, and avoid the need to do `*binding*` later